FeedStoreChallenge/Core Data/CoreDataFeedStore.swift

```diff
@@ -0,0 +1,121 @@
+//
+//  CoreDataFeedStore.swift
+//  FeedStoreChallenge
+//
+//  Created by Wilmer Barrios on 23/01/21.
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import Foundation
+import CoreData
+
+public final class CoreDataFeedStore: FeedStore {
+	
+	typealias StoredFeedData = (feed: [CDFeedImageItem], timestamp: Date)
+	
+	private let container: NSPersistentContainer
+	private let context: NSManagedObjectContext
+	private let modelName = "FeedStoreDataModel"
+	private let storeURL: URL!
+	
+	public init(storeURL: URL) throws {
+		
+		self.storeURL = storeURL
+		
+		guard let modelURL = Bundle(for: CoreDataFeedStore.self).url(forResource: modelName, withExtension:"momd"),
+			  let model = NSManagedObjectModel(contentsOf: modelURL) else {
+			throw NSError()
+		}
+		
+		let description = NSPersistentStoreDescription(url: storeURL)
+		container = NSPersistentContainer(name: modelName, managedObjectModel: model)
+		container.persistentStoreDescriptions = [description]
+		
+		var loadError: Swift.Error?
+		container.loadPersistentStores { loadError = $1 }
+		try loadError.map { throw $0 }
+		
+		context = container.newBackgroundContext()
+	}
+	
+	public func deleteCachedFeed(completion: @escaping DeletionCompletion) {
+		let context = self.context
+		context.perform {
+			let fetchRequest: NSFetchRequest<NSFetchRequestResult> = CDFeedImage.fetchRequest()
+			let batchDeleteRequest = NSBatchDeleteRequest(fetchRequest: fetchRequest)
+			do {
+				try context.execute(batchDeleteRequest)
+				completion(nil)
+			} catch let e {
+				completion(e)
+			}
+		}
+	}
+	
+	public func insert(_ feed: [LocalFeedImage], timestamp: Date, completion: @escaping InsertionCompletion) {
+		let context = self.context
+		context.perform {
+			let feed_data: [CDFeedImageItem] = feed.toCD(context: context)
+			let storeFeed = CDFeedImage(context: context)
+			storeFeed.feed_data = NSOrderedSet(array: feed_data)
+			storeFeed.feed_timestamp = timestamp
+			do {
+				try context.save()
+				completion(nil)
+			} catch let e {
+				completion(e)
+			}
+		}
+	}
+	
+	public func retrieve(completion: @escaping RetrievalCompletion) {
+		let context = self.context
+		context.perform {
+			let fetchRequest: NSFetchRequest<CDFeedImage> = CDFeedImage.fetchRequest()
```

Comment:
I believe the compiler can infer the type:

```suggestion
			let fetchRequest = CDFeedImage.fetchRequest()
```

Reply:
Hi Caio! Thanks for all your feed back.

This particular line give me compiler error, evidence attached.

![image](https://user-images.githubusercontent.com/40615926/105654953-b9278100-5e84-11eb-9a91-3793a3559d2a.png)